engine/src/connector.rs

```diff
@@ -134,16 +134,12 @@ impl EthConnectorContract {
     fn parse_event_message(&self, message: &str) -> TokenMessageData {
         let data: Vec<_> = message.split(':').collect();
         assert!(data.len() < 3);
+        let account_id = AccountId::try_from(data[0].as_bytes()).sdk_unwrap();
         if data.len() == 1 {
-            let account_id = data[0];
-            assert!(
-                is_valid_account_id(account_id.as_bytes()),
-                "ERR_INVALID_ACCOUNT_ID"
-            );
-            TokenMessageData::Near(account_id.into())
+            TokenMessageData::Near(account_id)
         } else {
             TokenMessageData::Eth {
-                address: data[0].into(),
+                address: account_id,
```

Comment:
Is this right? This is supposed to be an eth address (encoded in hex) isn't it, not a valid NEAR account ID?

Reply:
It is used only for:
```
            // Deposit to Eth accounts
            // fee is being minted in the `ft_on_transfer` callback method
            TokenMessageData::Eth { address, message } => {
                // Transfer to self and then transfer ETH in `ft_on_transfer`
                // address - is NEAR account
                let transfer_data = TransferCallCallArgs {
                    receiver_id: address,
                    amount: event.amount.as_u128(),
                    memo: None,
                    msg: self.set_message_for_on_transfer(event.fee, message),
                }
```

And after that it used for example here:
```
        self.ft
            .ft_transfer_call(&args.receiver_id, args.amount, &args.memo, args.msg);
```
That field should be `AccountId` type. Not `String`

Reply:
Seems like the field should be renamed if it is not actually an address.

Reply:
Yes, it is receiver_id. I'll rename it.